src/traces/scatter/hover.js

```diff
@@ -18,35 +18,44 @@ var fillHoverText = require('./fill_hover_text');
 var MAXDIST = Fx.constants.MAXDIST;
 
 module.exports = function hoverPoints(pointData, xval, yval, hovermode) {
-    var cd = pointData.cd,
-        trace = cd[0].trace,
-        xa = pointData.xa,
-        ya = pointData.ya,
-        xpx = xa.c2p(xval),
-        ypx = ya.c2p(yval),
-        pt = [xpx, ypx],
-        hoveron = trace.hoveron || '';
+    var cd = pointData.cd;
+    var trace = cd[0].trace;
+    var xa = pointData.xa;
+    var ya = pointData.ya;
+    var xpx = xa.c2p(xval);
+    var ypx = ya.c2p(yval);
+    var pt = [xpx, ypx];
+    var hoveron = trace.hoveron || '';
+    var minRad = (trace.mode.indexOf('markers') !== -1) ? 3 : 0.5;
 
     // look for points to hover on first, then take fills only if we
     // didn't find a point
     if(hoveron.indexOf('points') !== -1) {
         var dx = function(di) {
-                // scatter points: d.mrc is the calculated marker radius
-                // adjust the distance so if you're inside the marker it
-                // always will show up regardless of point size, but
-                // prioritize smaller points
+                // dx and dy are used in compare modes - here we want to always
+                // prioritize the closest data point, at least as long as markers are
+                // the same size or nonexistent, but still try to prioritize small markers too.
                 var rad = Math.max(3, di.mrc || 0);
-                return Math.max(Math.abs(xa.c2p(di.x) - xpx) - rad, 1 - 3 / rad);
```

Comment:
I think I copied this piece of logic elsewhere e.g. in [`scattergeo/hover.js`](https://github.com/plotly/plotly.js/blob/82f4c161cfa2035a4de18c0e2071ab7d6ab6f7f1/src/traces/scattergeo/hover.js#L45). Should we update it there too?

Reply:
scattergeo should be fine, as it only uses the `closest` form (ie real quadrature distance, not just x or just y). I suppose the `minRad` change is still relevant there, to better handle line-only traces, but that's it.